### PR TITLE
adding dbcolumn field to service/api layer

### PR DESCRIPTION
**Pull request by fanrukun**
adding dbcolumn field to service/api layer. 
1) in request/response of dataset resource, db columns fields exist in term of list of string maps.
2) internally in the service layer,  it is converted to List of DbTableColumn for processing.

### CODE REVIEW

File: src/main/java/marquez/common/models/DbColumn.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import static marquez.common.base.MorePreconditions.checkNotBlank;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import lombok.NonNull;
+import lombok.Value;
+
+@Value
+public class DbColumn {
```
Comment (wslulciuc) [inline src/main/java/marquez/common/models/DbColumn.java]: Given our _offline_ discussion, let's name this class as `Field`

File: src/main/java/marquez/common/models/DbColumn.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  String name;
```
Comment (wslulciuc) [inline src/main/java/marquez/common/models/DbColumn.java]: We'll want to define the following class that will replace these _String_ types:

- `FieldName` - Wraps the field name _String_
- `FieldType` - Defines the enum of allowed field types

File: src/main/java/marquez/common/models/DbColumn.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  @JsonCreator
+  public DbColumn(
+      @NonNull final String name, @NonNull final String type, @Nullable final String description) {
+    this.name = checkNotBlank(name);
+    this.type = checkNotBlank(type);
+    this.description = description;
+  }
+
+  public String getColumnUUID() {
```
Comment (wslulciuc) [inline src/main/java/marquez/common/models/DbColumn.java]: We'll only want versioning to be applied to classes that extend [`DatasetMeta`](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/service/models/DatasetMeta.java). Though this is a sweet helper method, let's keep versioning contained with `DatasetMeta.version()`

File: src/main/java/marquez/db/DatasetVersionDao.java [hunk 4]
```diff
@@ -38,6 +38,10 @@ default void insertAndUpdate(DatasetVersionRow row) {
       insert(row);
       createStreamVersionDao().insert((StreamVersionRow) row);
       createDatasetDao().update(row.getDatasetUuid(), row.getCreatedAt(), row.getVersion());
+    } else {
+      insert(row);
+      // todo: to complete by willy
```
Comment (wslulciuc) [inline src/main/java/marquez/db/DatasetVersionDao.java]: On it :)

File: src/main/java/marquez/api/models/DbTableRequest.java [hunk 5]
```diff
@@ -15,19 +15,30 @@
...
+import java.util.List;
 import javax.annotation.Nullable;
 import lombok.EqualsAndHashCode;
 import lombok.ToString;
+import marquez.common.models.DbColumn;
 
 @EqualsAndHashCode(callSuper = true)
 @ToString(callSuper = true)
 public final class DbTableRequest extends DatasetRequest {
+  private final List<DbColumn> columns;
```
Comment (wslulciuc) [inline src/main/java/marquez/api/models/DbTableRequest.java]: Given our _offline_ discussion, let's define `List<DbColumn>` as `List<Field>` and push the list to up to `DatasetRequest`

Comment (fanrukun) [reply src/main/java/marquez/api/models/DbTableRequest.java]: fixed.

File: src/main/java/marquez/common/models/DbColumn.java (hunk 1 above)
Comment (fanrukun) [reply src/main/java/marquez/common/models/DbColumn.java]: fixed.

File: src/main/java/marquez/common/models/DbColumn.java (hunk 3 above)
Comment (fanrukun) [reply src/main/java/marquez/common/models/DbColumn.java]: fixed.

File: src/main/java/marquez/common/models/FieldName.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+@EqualsAndHashCode
+@ToString
+public final class FieldName {
+  @Getter private final String value;
+
+  private FieldName(final String value) {
+    this.value = checkNotBlank(value, "value must not be blank");
+  }
+
+  public static FieldName of(final String value) {
```
Comment (fanrukun) [inline src/main/java/marquez/common/models/FieldName.java]: jsoncreator

File: src/main/java/marquez/service/models/DbTable.java [hunk 7]
```diff
@@ -15,22 +15,29 @@
...
+
   public DbTable(
       final DatasetName name,
       final DatasetName physicalName,
       final Instant createdAt,
       final Instant updatedAt,
       final SourceName sourceName,
-      @Nullable final String description) {
+      @Nullable final String description,
+      @Nullable final List<Field> fields) {
```
Comment (fanrukun) [inline src/main/java/marquez/service/models/DbTable.java]: move to Dataset

**Pull request merged**